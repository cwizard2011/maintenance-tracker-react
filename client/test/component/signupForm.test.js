import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import DefaultSignupForm, { SignupForm } from '../../src/components/SignupForm';

const mockStore = configureMockStore([thunk]);

const props = {
  register: jest.fn().mockResolvedValue(Promise.resolve()),
  userRegistration: {
    loading: false,
    error: {}
  }
};
const initialState = {
  authReducer: {
    user: {
      email: 'juliet',
      exp: 1540907850,
      iat: 1540886250,
      id: '48a698a0-1641-5aca-bc1b-de9b1a482ee1',
      user_role: 'user',
      username: 'juliet'
    },
    isAuthenticated: false
  }
};
const store = mockStore(initialState);

describe('Signup Form', () => {
  it('render signup form component correctly', () => {
    const wrapper = shallow(<SignupForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render connected component', () => {
    const wrapper = shallow(<DefaultSignupForm store={store} {...props} />);
    expect(wrapper.state()).toEqual({});
  });
  it('should call onSubmit', () => {
    const wrapper = shallow(<SignupForm {...props} />);
    const e = {
      preventDefault: jest.fn()
    };
    const instance = wrapper.instance();
    instance.onSubmit(e);
    expect(props.register).toHaveBeenCalled();
  });
  it('should set error in the state with password mismatch', () => {
    const wrapper = shallow(<SignupForm {...props} />);
    const e = {
      preventDefault: jest.fn()
    };
    wrapper.setState({
      password: '123',
      passwordConfirm: '1234'
    });
    const instance = wrapper.instance();
    instance.onSubmit(e);
    expect(props.register).toHaveBeenCalled();
    expect(wrapper.state().error).toBeTruthy();
  });
  it('should set server error in the state', () => {
    const newProps = {
      ...props,
      userRegistration: {
        error: {
          message: {
            errors: {
              firstname: ['Invalid'],
              lastname: ['Invalid'],
              email: ['Invalid'],
              password: ['Invalid'],
              username: ['Invalid'],
            }
          }
        }
      }
    };
    const wrapper = shallow(<SignupForm {...newProps} />);
    const e = {
      preventDefault: jest.fn()
    };
    const instance = wrapper.instance();
    instance.onSubmit(e);
    expect(props.register).toHaveBeenCalled();
    expect(wrapper.state().serverError).toBeTruthy();
  });
  it('should call onchange event', () => {
    const event = {
      target: {
        name: 'username',
        value: 'cwizard'
      }
    };
    const wrapper = shallow(<SignupForm {...props} />);
    const instance = wrapper.instance();
    instance.onChange(event);
    expect(wrapper.state().username).toEqual('cwizard');
  });
  it('should Render Loading when user submit login information', () => {
    const newProps = {
      ...props,
      userRegistration: {
        loading: true,
      }
    };
    const wrapper = shallow(<SignupForm {...newProps} />);
    expect(wrapper.find('Loading')).toBeTruthy();
  });
  it('should Display login error code', () => {
    const newProps = {
      ...props,
      userRegistration: {
        loading: false,
        error: {
          code: 409,
          message: {}
        },
      }
    };
    const wrapper = shallow(<SignupForm {...newProps} />);
    expect(wrapper.find('.red-text')).toBeTruthy();
  });
});
