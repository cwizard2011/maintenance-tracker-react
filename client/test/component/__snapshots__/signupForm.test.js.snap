// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Form render signup form component correctly 1`] = `
<div>
  <div />
  <h2
    className="center-heading"
  >
    Register
  </h2>
  <form
    id="signup-form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="First Name"
    >
      First Name
      <input
        name="firstname"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </label>
    <div
      className="red-text"
    />
    <label
      htmlFor="Last Name"
    >
      Last Name
      <input
        name="lastname"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </label>
    <div
      className="red-text"
    />
    <label
      htmlFor="Email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
    </label>
    <div
      className="red-text"
    />
    <label
      htmlFor="Username"
    >
      Username
      <input
        name="username"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </label>
    <div
      className="red-text"
    />
    <div
      className="red-text"
    />
    <label
      htmlFor="Password"
    >
      Password
      <input
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </label>
    <div
      className="red-text"
    />
    <label
      htmlFor="Confirm Password"
    >
      Confirm Password
      <input
        name="passwordConfirm"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </label>
    <div
      className="red-text"
      id="message"
    />
    <input
      className="right-btn btn"
      disabled={true}
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
